src/simularium/VisGeometry.ts

```diff
@@ -312,27 +320,28 @@ class VisGeometry {
         this.renderStyle = renderStyle;
 
         if (changed) {
-            this.constructInstancedFiberEndcaps();
+            // reset all agents of fiber type so their geometry is re-created.
+            for (const visAgent of this.visAgentInstances.values()) {
+                visAgent.visType = VisTypes.ID_VIS_TYPE_DEFAULT;
+            }
+
+            this.constructInstancedFibers();
         }
 
         this.updateScene(this.currentSceneAgents);
     }
 
-    private constructInstancedFiberEndcaps() {
+    private constructInstancedFibers() {
+        this.fibers.clear();
+        removeByName(this.instancedMeshGroup, InstancedFiberGroup.GROUP_NAME);
+
         // tell instanced geometry what representation to use.
         if (this.renderStyle === RenderStyle.GENERIC) {
-            this.fiberEndcaps = new InstancedFiberEndcapsFallback();
         } else {
-            this.fiberEndcaps = new InstancedFiberEndcaps();
+            this.fibers = new InstancedFiberGroup();
         }
```

Comment:
Looks like nothing happens after checking `if (this.renderStyle === RenderStyle.GENERIC)` anymore.